crates/runtime/src/dataconnector/duckdb.rs

```diff
@@ -89,12 +122,21 @@ impl DataConnector for DuckDB {
         &self,
         dataset: &Dataset,
     ) -> super::DataConnectorResult<Arc<dyn TableProvider>> {
-        Ok(
```

Comment:
I think it would be cleaner to keep this like it was `duckdb_factory` and just resolve the correct one to use in the `DataConnectorFactory::create`